### PR TITLE
all done

### PR DESCRIPTION
**Pull request by vlykhvar**
> OrderController

- Complete order - POST: /orders/complete?userId
- Get orders history for user - GET: /orders?userId

> AuthenticationController

- Register - POST: /register

> UserController

- Get user by email - GET: /users/by-email?email

> ShoppingCartController

- Add movie session - POST: /shopping-carts/movie-sessions?userId&movieSessionId
- Get by user - GET: /shopping-carts/by-user?userId

### CODE REVIEW

File: src/main/java/com/dev/theater/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                           UserService userService,
+                           OrderMapper orderMapper,
+                           ShoppingCartService shoppingCartService) {
+        this.orderService = orderService;
+        this.userService = userService;
+        this.orderMapper = orderMapper;
+        this.shoppingCartService = shoppingCartService;
+    }
+
+    @GetMapping("/orders")
```
Comment (Likh-Alex) [inline src/main/java/com/dev/theater/controller/OrderController.java]: not necessary to write specific `endpoint` as the same `url` is entered in the `RequestMapping` annotation

**Review by Likh-Alex — APPROVED** [review]

File: src/main/java/com/dev/theater/model/dto/OrderResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package com.dev.theater.model.dto;
+
+import com.dev.theater.model.Ticket;
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long id;
+    private String orderTime;
+    private List<Ticket> tickets;
```
Comment (boroda4436) [inline src/main/java/com/dev/theater/model/dto/OrderResponseDto.java]: Nope. Do not expose models from the controller layer.
@Likh-Alex FYI

File: src/main/java/com/dev/theater/model/dto/ShoppingCartResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package com.dev.theater.model.dto;
+
+import com.dev.theater.model.Ticket;
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Ticket> tickets;
```
Comment (boroda4436) [inline src/main/java/com/dev/theater/model/dto/ShoppingCartResponseDto.java]: same here

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/theater/model/dto/ShoppingCartResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package com.dev.theater.model.dto;
+
+import com.dev.theater.model.Ticket;
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Ticket> ticketIds;
```
Comment (boroda4436) [inline src/main/java/com/dev/theater/model/dto/ShoppingCartResponseDto.java]: Nope. Please see Monday's QnA video. You should use a list of ids here

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/theater/mapper/OrderMapper.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    public OrderResponseDto orderToDto(Order order) {
+        OrderResponseDto orderResponseDto = new OrderResponseDto();
+        orderResponseDto.setId(order.getId());
+        orderResponseDto.setOrderTime(order.getOrderTime().toString());
+        List<Ticket> listOld = order.getTickets();
+        List<Long> responseList = new ArrayList<>();
+        for (Ticket ticket : listOld) {
+            responseList.add(ticket.getId());
+        }
+        orderResponseDto.setTicketIds(responseList);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/mapper/OrderMapper.java]: let's use Stream API?

File: src/main/java/com/dev/theater/mapper/ShoppingCartMapper.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    public ShoppingCartResponseDto shoppingCartToDto(ShoppingCart shoppingCart) {
+        ShoppingCartResponseDto shoppingCartResponseDto = new ShoppingCartResponseDto();
+        shoppingCartResponseDto.setId(shoppingCart.getId());
+        shoppingCartResponseDto.setUserEmail(shoppingCart.getUser().getEmail());
+        List<Ticket> listOld = shoppingCart.getTicketList();
+        List<Long> responseList = new ArrayList<>();
+        for (Ticket ticket : listOld) {
+            responseList.add(ticket.getId());
+        }
+        shoppingCartResponseDto.setTicketIds(responseList);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/theater/mapper/ShoppingCartMapper.java]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**